packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -877,7 +890,10 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with Restoratio
     final bool enabled = widget.enabled ?? true;
     final Offset cursorOffset = Offset(_iOSHorizontalCursorOffsetPixels / MediaQuery.of(context)!.devicePixelRatio, 0);
     if (widget.maxLength != null && widget.maxLengthEnforced) {
-      formatters.add(LengthLimitingTextInputFormatter(widget.maxLength));
+      formatters.add(LengthLimitingTextInputFormatter(
```

Comment:
I tried `maxLengthEnforced || maxLengthEnforcement != warningOnly` but it won't work if the user manually set the `maxLengthEnforced`, since it's `true` by default. Maybe we should remove it entirely.

Reply:
Something like:
```dart
    if (widget.maxLength != null && widget.maxLengthEnforced) {
      switch (widget.maxLengthEnforced) {
         case .warningOnly:
           assert(false, '....');
          case .enforced:
          case .composingUnenforced:
            formatters.add(...);
       }
    }
```